pkg/storagev2/inmemstore/inmemstore.go

```diff
@@ -105,6 +119,15 @@ func (s *Store) Delete(i storage.Item) error {
 	return nil
 }
 
+// for batching purpose
+func (s *Store) delete(key string) error {
+	if _, deleted := s.st.Delete(key); !deleted {
+		return storage.ErrNotFound
```

Comment:
Returning error when attempting to remove item which is not present in store may not be ideal design decision. Handing this error will propagate all the way up to user which form their perspective can be vary valid case.

Error should indicate that operation was not performed successfully and not that zero elements were affected.

Reply:
how should this be refactored?

Reply:
In this case of `delete` and `Delete` methods no error should be returned if nothing was removed.

Reply:
I don't think this is the right way to go about it. The error should be returned and the user can choose how to handle it. He can very well ignore it. But if there is a case when the user cares about this, it should be sent up.

Reply:
It add unnecessary complexity because user need to check if error is because operation has failed or nothing was removed. 

After executing `delete` operation all what user should care is that this element is not present in the state, considering operation was successful.

What should happen happen when this error was returned when batch is committed? Should whole batch fail? Since user can't deal with this error, can't ignore it, their whole batch is going to fail unwillingly. It looks like we made decision for user in this case